hathor/sysctl/sysctl.py

```diff
@@ -0,0 +1,103 @@
+# Copyright 2023 Hathor Labs
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, Callable, Dict, Iterator, NamedTuple, Optional, Set, Tuple
+
+from hathor.sysctl.exception import SysctlEntryNotFound, SysctlReadOnlyEntry, SysctlWriteOnlyEntry
+
+Getter = Callable[[], Any]
+Setter = Callable[..., None]
+
+
+class SysctlMatch(NamedTuple):
+    sysctl: 'Sysctl'
+    path: str
+
+
+class Sysctl:
+    """A node in the sysctl tree."""
+
+    def __init__(self) -> None:
+        self._children: Dict[str, 'Sysctl'] = {}
+        self._getters: Dict[str, Getter] = {}
+        self._setters: Dict[str, Setter] = {}
+        self._paths: Set[str] = set()
+
+    def put_child(self, path: str, sysctl: 'Sysctl') -> None:
+        """Add a child to the tree."""
+        assert path not in self._children
+        assert path not in self._paths
+        self._children[path] = sysctl
+
+    def register(self, path: str, getter: Optional[Getter], setter: Optional[Setter]) -> None:
```

Comment:
I might have misunderstood how the `children` attribute works but you have in the manager this code:

```python
self.register('rate_limit.global.send_tips', ...)
```

Shouldn't you handle this path registration with dots in this method, in order to create the children for each partition?

Reply:
I don't think it's worth it. I left to the coder to decide whether it's better to add a branch in the tree or just add the remaining path to the current node. As an example, in the p2p part, we have the commands `always_enable_sync` and `always_enable_sync.readtxt`, both in the same node. What do you think of this design?

Reply:
I think it's good